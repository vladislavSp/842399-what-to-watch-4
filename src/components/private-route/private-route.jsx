import React from "react";
import PropTypes from "prop-types";
import {Route, Redirect} from "react-router-dom";
import {getAuthorizationStatus} from "../../reducer/user/selectors";
import {connect} from "react-redux";

export const PrivateRoute = (props) => {
  const {
    render,
    path,
    exact,
    computedMatch,
    authorizationStatus,
    requiredAuthStatus,
    pathToRedirect
  } = props;
  return (
    <Route
      path={path}
      exact={exact}
      match={computedMatch}
      render={() => {
        return authorizationStatus === requiredAuthStatus ? (
          render(computedMatch)
        ) : (
          <Redirect to={pathToRedirect} />
        );
      }}
    />
  );
};

PrivateRoute.propTypes = {
  authorizationStatus: PropTypes.string.isRequired,
  exact: PropTypes.bool.isRequired,
  path: PropTypes.string.isRequired,
  render: PropTypes.func.isRequired,
  computedMatch: PropTypes.object,
  requiredAuthStatus: PropTypes.string,
  pathToRedirect: PropTypes.string
};

const mapStateToProps = (state) => ({
  authorizationStatus: getAuthorizationStatus(state)
});

export default connect(mapStateToProps)(PrivateRoute);
