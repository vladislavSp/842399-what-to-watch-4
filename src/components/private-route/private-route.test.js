import React from "react";
import renderer from "react-test-renderer";
import {PrivateRoute} from "./private-route.jsx";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {Router} from "react-router-dom";
import history from "../../history";
import MyList from "../my-list/my-list.jsx";

const mockStore = configureStore([]);
const store = mockStore({});

jest.mock(`../my-list/my-list`, () => `MyList`);

it(`Private route renders correctly`, () => {
  const tree = renderer
    .create(
        <Router history={history}>
          <Provider store={store}>
            <PrivateRoute
              exact
              path="/mylist"
              requiredAuthorizationStatus={`AUTH`}
              pathToRedirect={`/login`}
              authorizationStatus={`AUTH`}
              render={() => {
                return <MyList />;
              }}
            />
          </Provider>
        </Router>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
