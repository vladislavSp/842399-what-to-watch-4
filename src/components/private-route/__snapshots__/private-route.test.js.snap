// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Private route renders correctly 1`] = `null`;
